File: src/main/java/io/myosotisdev/minestom/command/manage/ServerSubCommand.java
package io.myosotisdev.minestom.command.manage;

import io.myosotisdev.minestom.Minestom;
import io.myosotisdev.minestom.command.AbstractCommand;
import io.myosotisdev.minestom.permission.Permission;
import io.myosotisdev.utopianism.util.Namespaces;

public class ServerSubCommand extends AbstractCommand
{
    public static final String Name = "server";
    public static final String[] Aliases = new String[0];

    public ServerSubCommand(AbstractCommand parent)
    {
        super(parent, Name);

        this.addSubcommand(new Stop(this));
    }

    @Override
    public Permission permission()
    {
        return Permission.ofString(Namespaces.Minestom, "manage.server");
    }

    static class Stop extends AbstractCommand
    {
        static final String Name = "stop";

        public Stop(AbstractCommand command)
        {
            super(command, Name);

            addSyntax((sender, context) -> {
                if(!Minestom.checkPermission(sender, permission()))
                    sender.sendMessage("You don't have permission to do that!");
                else
                {
                    Minestom.stop();
                }
            });
        }

        @Override
        public Permission permission()
        {
            return getParent().permission();
        }
    }
}

File: src/main/java/io/myosotisdev/minestom/command/manage/ServerSubCommand.kt
package io.myosotisdev.minestom.command.manage

import io.myosotisdev.minestom.Minestom.stop
import io.myosotisdev.minestom.command.AbstractCommand
import io.myosotisdev.minestom.permission.Permission
import io.myosotisdev.minestom.permission.Permission.Companion.ofString
import io.myosotisdev.utopianism.util.Namespaces

class ServerSubCommand(parent: AbstractCommand?) : AbstractCommand(parent, Name)
{
    companion object
    {
        const val Name = "server"
        val Aliases = arrayOfNulls<String>(0)
    }

    init
    {
        addSubcommand(Stop(this))
    }

    override fun permission(): Permission?
    {
        return ofString(Namespaces.Minestom, "manage.server")
    }

    internal class Stop(command: AbstractCommand?) : AbstractCommand(command, Name)
    {
        companion object { const val Name = "stop" }

        init
        {
            addSyntax({ sender, context ->
                stop()
            })
        }

        override fun permission(): Permission?
        {
            return parent!!.permission()
        }
    }
}